Adds Title class with methods to set page titles, and sets titles for pages

SigmasonicX opened the pull request:
- Excludes work sections and docs pages
- Excludes Dashboard

Issues:
- Unable to get titles for work sections and new sections working properly
- When refresh page, it first displays the title as the default "Offprint" and then correctly sets the title.

SigmasonicX committed:
Adds GlobalMethod class with methods to set page titles
- Sets titles for some pages

SigmasonicX committed:
Adds GlobalMethod class with methods to set page titles
- Sets titles for some pages

SigmasonicX committed:
Adds GlobalMethod class with methods to set page titles
- Sets titles for some pages

Figments committed:
moved queue service functions to queue controller

SigmasonicX committed:
Add titles for most pages
- Excludes work sections and docs pages
- Excludes Dashboard

SigmasonicX committed:
Add titles for most pages
- Excludes work sections and docs pages
- Excludes Dashboard

SigmasonicX committed:
Merge branch 'master' of github.com:SigmasonicX/pulp-fiction

# Conflicts:
#	packages/client/src/app/shared/global-constants.ts

pingzing commented on packages/client/src/app/shared/global-constants.ts:
```diff
@@ -1,3 +1,26 @@
 export class GlobalConstants {
```
I think you can get away with just calling this `Constants`; I'd say the "Global" is implied.

pingzing commented on packages/client/src/app/shared/global-methods.ts:
```diff
@@ -0,0 +1,16 @@
+import { GlobalConstants } from './global-constants';
+
+export class GlobalMethods {
```
Two things:

First:
Personal opinion: I'm not a fan of "Util", "Helper", etc rollup files. Names should be descriptive. In this case, I think something like `titles.ts` for the filename and `Titles` for the class name would be more informative here.

Second:
These functions don't actually need to live in a class--the more idiomatic way to define them would be as naked functions, e.g.
```typescript
export function setDefaultTitle(): void {
    // etc
} 
```

If you want the benefit of things being "inside" something else, you can use a namespace, instead!

```typescript
namespace Titles {
  export function setDefaultTitle(): void {
    // etc
  }
}
```

...and just do `import Titles` elsewhere.

SigmasonicX committed:
- Change GlobalConstants to Constants
- Change GlobalMethods to Title
- Add Title to index

SigmasonicX committed:
- Change GlobalConstants to Constants
- Change GlobalMethods to Title
- Add Title to index

SigmasonicX committed:
Change Title to namespace

SigmasonicX committed:
Change Title to namespace

Figments committed:
backend work for clientside news page done

Figments committed:
Merge branch 'master' into feature/client-news

Figments committed:
fixed blogs and works pages not working

Figments committed:
added frontend news service

Figments committed:
added news page resolver

Figments committed:
added post page component

Figments committed:
changed news page resolver to reflect component name

Figments committed:
added post page to routing

Figments committed:
added functions to news component

Figments committed:
added toolbar styles from dashboard

SigmasonicX committed:
Use correctly formatted username for portfolio subpages

SigmasonicX committed:
Use correctly formatted username for portfolio subpages

SigmasonicX committed:
Set title for collection pages

SigmasonicX committed:
Set title for collection pages

Figments committed:
added news feed resolver

SigmasonicX committed:
- Set titles for work sections
- Changes file name of global-constants to constants

SigmasonicX committed:
- Set titles for work sections
- Changes file name of global-constants to constants

Figments committed:
added news feed to client

Figments committed:
pagination now accomplished via resolver

Figments committed:
made post route into its own parent

Figments committed:
added author and post links to news feed

Figments committed:
moved post page to root of pages

Figments committed:
added post page

Figments committed:
added link to profile in post

Figments committed:
added left margin to lists

Figments committed:
added top margin

Figments committed:
added backend routes for content comments

Figments committed:
modified comments component to accept content comments

Figments committed:
added comments to news posts

Figments committed:
added ability for content to have incrementing views and comments counters

Commit:
Fix image styling, improve other style issues
 - Allow users to set image alignment
 - Improve HTML sanitizer to allow image resizing, and image alignment
 - Add font size and font color options to editor
 - Improve HTML sanitizer to allow font sizes and colors

Commit:
Remove stray empty entry in figure allowed classes

Commit:
Remove unused imports

pingzing committed:
Merge pull request #189 from OffprintStudios/fix/body-image-styling

Fix image styling, improve other style issues

Figments committed:
added global style class for empty pages

Figments committed:
changed comments service to not use .then() calls on save

pingzing commented on packages/client/src/app/shared/title.ts:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    export function setThreePartTitle(title1: string, title2: string): void {
+        document.title = title1 + Constants.SEPARATOR + title2 +
```
suggestion/idea/thought: Javascript/TypeScript support string interpolation, so you could also write this as
```typescript
document.title = `${title1} ${Constants.SEPARATOR}$ ${title2} ${Constants.SEPARATOR} ${Constants.OFFPRINT}`;
```
...which is a little hard to read, and (personal opinion) I think you're better off not defining a separator constant anyway. Then you could do:
```typescript
document.title = `${title1} - ${title2} - ${Constants.OFFPRINT}` ;
```

Figments committed:
published check is implicit

Figments committed:
Merge branch 'master' into feature/client-news

Figments committed:
Merge pull request #188 from OffprintStudios/feature/client-news

feature/client-news

SigmasonicX committed:
Merge branch 'master' of github.com:SigmasonicX/pulp-fiction

# Conflicts:
#	packages/client/src/app/pages/news/news.component.ts
